src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataAnalysis.cs

```diff
@@ -27,12 +32,13 @@ private TaintedDataAnalysis(TaintedDataOperationVisitor operationVisitor)
             TaintedDataSymbolMap<SourceInfo> taintedSourceInfos,
             TaintedDataSymbolMap<SanitizerInfo> taintedSanitizerInfos,
             TaintedDataSymbolMap<SinkInfo> taintedSinkInfos,
-            CancellationToken cancellationToken)
+            CancellationToken cancellationToken,
+            bool needValueContentAnalysis = false)
```

Comment:
>bool needValueContentAnalysis = false [](start = 12, length = 37)

Instead of this being specified by the caller, we should figure this out based on whether any of the taintedSourceInfos / taintedSanitizerInfos / taintedSinkInfos need value content analysis.

TaintedDataAnalysis results are intended to be used by multiple analyzers, if the analyzers share the same sources and sanitizers.

Reply:
+1

Reply:
I checked whether it needs ValueContentAnalysis with related taintedSourceInfos in the analyzer then pass it to this method. You mean move the check to TaintedDataAnalysis?

There are 2 reasons that I put needValueContentAnalysis in the caller:
1. One rootOperation doesn't cover all taintedSourceInfos / taintedSanitizerInfos / taintedSinkInfos, current rootOperation may doesn't need ValueContentAnalysis while other taintedSourceInfos that don't belong to this rootOperation may need. If I don't pass needValueContentAnalysis from the caller, I need to pass the related taintedSourceInfos to this method. Otherwise, it may does some no need ValueContentAnalysis.
2. If the SourceInfo needs ValueContentAnalysis, I need to check if the value meets requirement first then to decide if put the operation into rootOperationsNeedingAnalysis. Now that I get the result whether need ValueContentAnalysis, why I pass the related taintedSourceInfos?